borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
With this change, yes. Before: depends. With the typical write structure borg does (write manifest, commit) and the old compact algorithm commit segments would practically always be compacted. With the new compact algorithm it would depend on whether there is enough other superseded data in the segment containing a commit (a couple-kilobytes manifest doesn't do the trick).

Plan: merge this as is; amend the compact algorithm after the other compact PR to consider commits as well, so these little files get deleted as well.